crates/router/src/services/api.rs

```diff
@@ -1932,6 +1931,38 @@ pub fn build_payment_link_html(
 
     context.insert("logging_template", &logging_template);
 
+    Ok((tera, context))
+}
+
+pub fn build_payment_link_html(
+    payment_link_data: PaymentLinkFormData,
+) -> CustomResult<String, errors::ApiErrorResponse> {
+    let (tera, mut context) = build_payment_link_template(payment_link_data)
+        .attach_printable("Failed to build payment link's HTML template")?;
+    let payment_link_initiator =
+        include_str!("../core/payment_link/payment_link_initiate/payment_link_initiator.js")
+            .to_string();
+    context.insert("payment_link_initiator", &payment_link_initiator);
+
+    match tera.render("payment_link", &context) {
+        Ok(rendered_html) => Ok(rendered_html),
+        Err(tera_error) => {
+            crate::logger::warn!("{tera_error}");
+            Err(errors::ApiErrorResponse::InternalServerError)?
```

Comment:
Attach context messages using `attach_printable` or `attach_printable_lazy` whenever raising internal server errors. 

You can also use `change_context()` here instead?

```suggestion
            tera_error
                .change_context(errors::ApiErrorResponse::InternalServerError)
                .attach_printable("...")
```